yt/frontends/ramses/data_structures.py

```diff
@@ -109,37 +105,62 @@ def check_standard_files(folder, iout):
         ok &= (folder / f"info_{iout}.txt").is_file()
         return ok
 
+    @staticmethod
+    def _match_output_and_group(
+        path: Path,
+    ) -> Tuple[Path, Optional[Path], Optional[str]]:
+
+        # Three cases for path
+        # 1. path == output_XXXXX and group_00001 exists
+        # 2. path == output_XXXXX and group_00001 does not exist
+        # 3. path == output_XXXXX/group_YYYYYY
+        iout_match = OUTPUT_DIR_RE.match(path.name)
+        if iout_match:
+            group_dir = path / "group_00001"
+
+            return path, group_dir if group_dir.is_dir() else None, iout_match.group(1)
+        else:
+            iout_match = OUTPUT_DIR_RE.match(path.parent.name)
+
+            return path.parent, path, iout_match.group(1) if iout_match else None
+
     @classmethod
-    def test_with_folder_name(cls, output_dir):
-        iout_match = OUTPUT_DIR_RE.match(output_dir.name)
-        ok = output_dir.is_dir() and iout_match is not None
+    def test_with_folder_name(
+        cls, output_dir: Path
+    ) -> Tuple[bool, Optional[Path], Optional[Path], Optional[Path]]:
+        output_dir, group_dir, iout = cls._match_output_and_group(output_dir)
+        ok = output_dir.is_dir() and iout is not None
+
+        info_fname: Optional[Path]
+
         if ok:
-            iout = iout_match.group(2)
-            ok &= cls.check_standard_files(output_dir, iout)
-            info_fname = output_dir / f"info_{iout}.txt"
+            ok &= cls.check_standard_files(group_dir or output_dir, iout)
+            info_fname = (group_dir or output_dir) / f"info_{iout}.txt"
```

Comment:
`parent_dir` is more of a place-holder here, I'd suggest finding a better name
```suggestion
            parent_dir = group_dir or output_dir
            ok &= cls.check_standard_files(parent_dir, iout)
            info_fname = parent_dir / f"info_{iout}.txt"
```